qiskit/ignis/experiments/calibration/methods/cross_resonance.py

```diff
@@ -0,0 +1,94 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Data source to generate schedule."""
+
+
+from typing import Optional, List, Tuple
+
+from qiskit.ignis.experiments.calibration import CircuitBasedGenerator
+from qiskit.ignis.experiments.calibration.data_processing import DataProcessingSteps
+from qiskit.ignis.experiments.calibration.cal_base_analysis import BaseCalibrationAnalysis
+from qiskit.ignis.experiments.calibration.instruction_data import InstructionsDefinition
+from qiskit.ignis.experiments.calibration.cal_base_experiment import BaseCalibrationExperiment
+from qiskit.ignis.experiments.calibration.analysis.trigonometric import CosinusoidalFit
+
+from qiskit.pulse import ControlChannel
+
+
+class RoughCRAmplitude(BaseCalibrationExperiment):
+    """Performs a rough amplitude calibration by scanning the amplitude of the pulse."""
+
+    def __init__(self,
+                 inst_def: InstructionsDefinition,
+                 qubits: Tuple[int, int],
+                 data_processing: DataProcessingSteps,
+                 amp_vals: List,
+                 cr_name: Optional[str] = 'cr',
+                 calibration_group: Optional[str] = 'default',
+                 analysis_class: Optional[BaseCalibrationAnalysis] = None,
+                 job: Optional = None):
+        """
+        The amplitude of the pulses on the u channel are scanned. The echoed-cr gates with cr90p
+        and cr90m will have amplitudes scanned with opposite signs.
+
+        Args:
+            inst_def: The class that defines the instructions for this calibration.
+            qubits: Qubits of the gate given as (Control, Target).
+            data_processing: Steps used to process the data from the Result.
+            amp_vals: Amplitude values to scan in the calibration.
+            analysis_class: Analysis class used.
+            job: Optional job id to retrieve past experiments.
+            cr_name: Name of the cross-resonance gate from the instructions definition to use.
+        """
+        # todo calibration_group is not handled yet
+        # todo get qubit property from other database.
+        # channel ref frequency is different from pulse sideband and thus
+        # this value is stored in another relational database.
+        # something like qubit property table where f01, anharmonicity, T1, T2, etc... exist.
+        freq01 = None
+
+        u_ch_name = inst_def.get_channel_name(qubits, ControlChannel)
```

Comment:
can we return channel object rather than name? when do we need to get just name?

Reply:
Updated. This now returns channels.